From: Robert Mader <robert.mader@posteo.de>
Date: Thu, 18 Mar 2021 13:42:30 +0100
Subject: region-utils: Fix typo in crop_and_scale() fastpath

Fixes 09b1bbb1cf64ec5e90396d4cedd4bbd84df899a1

(cherry picked from commit 1bfd932f15e57583fe03ae26bce23663e9003ae8)

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1808>
---
 src/compositor/region-utils.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compositor/region-utils.c b/src/compositor/region-utils.c
index 02415ce..ac4048a 100644
--- a/src/compositor/region-utils.c
+++ b/src/compositor/region-utils.c
@@ -428,8 +428,8 @@ meta_region_crop_and_scale (cairo_region_t  *region,
     {
       viewport_region = cairo_region_copy (region);
 
-      if (G_APPROX_VALUE (src_rect->origin.x, 0, FLT_EPSILON) ||
-          G_APPROX_VALUE (src_rect->origin.y, 0, FLT_EPSILON))
+      if (!G_APPROX_VALUE (src_rect->origin.x, 0, FLT_EPSILON) ||
+          !G_APPROX_VALUE (src_rect->origin.y, 0, FLT_EPSILON))
         {
           cairo_region_translate (viewport_region,
                                   (int) src_rect->origin.x,
